src/hooks/useNetworkSwitch.ts

```diff
@@ -17,25 +18,50 @@ export type UseNetworkSwitchProps = {
 export const useNetworkSwitch = ({ onSelectNetworkCallback }: UseNetworkSwitchProps = {}) => {
   const { connector, chainId, account } = useActiveWeb3React()
   const unsupportedChainIdError = useUnsupportedChainIdError()
+  const navigate = useNavigate()
+  const { pathname } = useLocation()
 
   const selectNetwork = useCallback(
-    (optionChainId?: ChainId) => {
+    async (optionChainId?: ChainId) => {
       if (optionChainId === undefined || (optionChainId === chainId && !unsupportedChainIdError)) return
+
+      let changeChainId: Promise<unknown> | undefined
       if (!account && !unsupportedChainIdError && connector instanceof CustomNetworkConnector)
         connector.changeChainId(optionChainId)
       else if (!account && unsupportedChainIdError && connector instanceof CustomNetworkConnector)
-        connector.switchUnsupportedNetwork(NETWORK_DETAIL[optionChainId])
+        changeChainId = connector.switchUnsupportedNetwork(NETWORK_DETAIL[optionChainId])
       else if (connector instanceof InjectedConnector)
-        switchOrAddNetwork(NETWORK_DETAIL[optionChainId], account || undefined)
+        changeChainId = switchOrAddNetwork(NETWORK_DETAIL[optionChainId], account || undefined)
       else if (connector instanceof CustomWalletLinkConnector)
-        connector.changeChainId(NETWORK_DETAIL[optionChainId], account || undefined)
+        changeChainId = connector.changeChainId(NETWORK_DETAIL[optionChainId], account || undefined)
 
       if (onSelectNetworkCallback) onSelectNetworkCallback()
+      try {
+        if (changeChainId) await changeChainId
+        unavailableRedirect(optionChainId, navigate, pathname)
+      } catch (e) {
+        throw e
```

Comment:
Are we catching this error above?

Reply:
I thought the promise was throwing an exception when we cancelled the operation. That is not the case. I remove this try catch and am now using an IF statement to check if we need to redirect